bapt/__init__.py

```diff
@@ -112,3 +113,83 @@ def read_config(filename):
+                   name_colour='w', fade_cb=False):
+
+    width = (bar_width/2. + gap/2.) * len(data) if not width else width
+    emin = emin if emin else min([d['vbo'] for d in data]) - 1
+    emax = emax if emax else max([d['vbo'] + d['eg'] for d in data]) + 1
+
+    plt = pretty_plot(width=width, height=height, plt=plt, fonts=[font])
+    ax = plt.gca()
+
+    pad = -0.4
+    for i, compound in enumerate(data):
+        x = i * (bar_width + gap)
+        ip = compound['vbo']
+        ea = compound['vbo'] + compound['eg']
+
+        fade = 'fade' in compound and compound['fade']
+        edge_c_cb = '#808080' if fade or fade_cb else 'k'
+        edge_c_vb = '#808080' if fade else 'k'
+        edge_z = 4 if fade else 5
+
+        vc = vb_cmap if 'vb_gradient' not in compound else \
+            compound['vb_gradient']
+        cc = cb_cmap if 'cb_gradient' not in compound else \
+            compound['cb_gradient']
+
+        gbar(ax, x, ip, bottom=emin, bar_width=bar_width, show_edge=True,
+             gradient=vc, edge_colour=edge_c_vb, edge_zorder=edge_z)
+        gbar(ax, x, ea, bottom=emax, bar_width=bar_width, show_edge=True,
+             gradient=cc, edge_colour=edge_c_cb, edge_zorder=edge_z)
+
+        dashed_arrow(ax, x + bar_width/6., ip - pad/3, 0,
+                     ea-ip + 2 * pad/3, colour='k', line_width=_linewidth)
+        ax.text(x + bar_width/4., ip + pad/2 + compound['eg']/2,
+                '{:.2f} eV'.format(compound['eg']), ha='left',
+                va='bottom', size=label_size, color='k', zorder=2)
+
+        ax.text(x + bar_width/2., ip + pad/2, compound['name'], zorder=2,
+                ha='center', va='top', size=label_size, color=name_colour)
+        if i > 0:
+            if not hide_vbo:
+                ax.text(x + bar_width/2., ip + 1.65*pad,
+                        f"VBO: {compound['vbo'] - data[i-1]['vbo']:+.2f} eV", zorder=2,
+                        ha='center', va='top', size=label_size-3, color=name_colour,
+                        fontfamily='serif')
+            if not hide_cbo:
+                ax.text(x + bar_width/2., ea - pad,
+                        f"CBO: {compound['cbo'] - data[i-1]['cbo']:+.2f} eV", zorder=2,
+                        ha='center', va='top', size=label_size-3, color=name_colour,
+                        fontfamily='serif')
+
+        if fade:
+            fadebar(ax, x, emin, bar_width=bar_width, bottom=emax)
+        elif fade_cb:
+            fadebar(ax, x, ea, bar_width=bar_width, bottom=emax, zorder=1)
+
+    ax.set_ylim((emin, emax))
+    ax.set_xlim((0, (len(data) * bar_width) + ((len(data) - 1) * gap)))
+    ax.set_xticks([])
+
+    if show_axis:
+        ax.set_ylabel("Energy (eV)", size=label_size)
+        ax.yaxis.set_major_locator(MultipleLocator(1))
+        for spine in ax.spines.values():
+            spine.set_zorder(5)
+        ax.tick_params(which='major', width=1.75*_linewidth) # prettify
```

Comment:
Is the width controlling the thickness of the line? In which case, can you keep the line thickness the same as all other axis lines for consistency.

Reply:
Yeah this was just to change the major axis tick linewidth to make it obvious that the zero of energy was set to the VBM of the first compound:  
To give this:

![image](https://user-images.githubusercontent.com/51478689/72367065-69e83580-36f3-11ea-8c1b-1eeb1c25c56e.png)

Instead of this:

![image](https://user-images.githubusercontent.com/51478689/72367108-84221380-36f3-11ea-8d3e-8023a7a235e3.png)

Either way, I suppose it's clear enough without changing the width, and looks a bit cleaner and more consistent, without changing it, so I'll change it now.